packages/zoe/state.js

```diff
@@ -1,190 +1,259 @@
-          purseP,
-          extentOpsDescP,
-        ]);
-
-        assayToDescOps.init(assay, unitOps);
-        assayToLabel.init(assay, label);
-        assayToPurse.init(assay, purse);
-        const { name, extentOpArgs = [] } = extentOpsDesc;
-        assayToExtentOps.init(assay, extentOpsLib[name](...extentOpArgs));
-      }
-      return harden({
-        unitOps: assayToDescOps.get(assay),
-        label: assayToLabel.get(assay),
-        purse: assayToPurse.get(assay),
-        extentOps: assayToExtentOps.get(assay),
+  const table = harden({
+    validate: validateFn,
+    create: (handle, record) => {
+      record = harden({
+        ...record,
+        handle, // reliably add the handle to the record
       });
+      table.validate(record);
+      handleToRecord.init(handle, record);
+      return record;
     },
-    recordOffer: (offerHandle, offerRules, extents, assays, result) => {
-      const { payoutRules, exit } = offerRules;
-      offerHandleToExtents.init(offerHandle, extents);
-      offerHandleToAssays.init(offerHandle, assays);
-      offerHandleToPayoutRules.init(offerHandle, payoutRules);
-      offerHandleToExitRule.init(offerHandle, exit);
-      offerHandleToResult.init(offerHandle, result);
-      activeOffers.add(offerHandle);
-    },
-    recordUsedInInstance: (instanceHandle, offerHandle) =>
-      offerHandleToInstanceHandle.init(offerHandle, instanceHandle),
-    getInstanceHandleForOfferHandle: offerHandle => {
-      if (offerHandleToInstanceHandle.has(offerHandle)) {
-        return offerHandleToInstanceHandle.get(offerHandle);
-      }
-      return undefined;
-    },
-    setExtentsFor: (offerHandles, reallocation) =>
-      offerHandles.map((offerHandle, i) =>
-        offerHandleToExtents.set(offerHandle, reallocation[i]),
-      ),
-    getResultsFor: offerHandles =>
-      offerHandles.map(offerHandle => offerHandleToResult.get(offerHandle)),
-    removeOffers: offerHandles => {
-      // has-side-effects
-      // eslint-disable-next-line array-callback-return
-      offerHandles.map(offerHandle => {
-        offerHandleToExtents.delete(offerHandle);
-        offerHandleToAssays.delete(offerHandle);
-        offerHandleToPayoutRules.delete(offerHandle);
-        offerHandleToExitRule.delete(offerHandle);
-        offerHandleToResult.delete(offerHandle);
-        if (offerHandleToInstanceHandle.has(offerHandle)) {
-          offerHandleToInstanceHandle.delete(offerHandle);
-        }
-      });
-    },
-    setOffersAsInactive: offerHandles => {
-      offerHandles.map(offerHandle => activeOffers.delete(offerHandle));
+    get: handleToRecord.get,
+    has: handleToRecord.has,
+    delete: handleToRecord.delete,
+    update: (handle, partialRecord) => {
+      const record = handleToRecord.get(handle);
+      const updatedRecord = {
+        ...record,
+        ...partialRecord,
+      };
+      table.validate(updatedRecord);
+      handleToRecord.set(handle, updatedRecord);
+      return record;
```

Comment:
I looked, and in most cases, we only want the handle. There is one case in which we return more than the handle, but it will change with the invites PR.  I changed this to only return the handle with a comment explaining the unusual case and how it will change in the future.

Reply:
Thanks for catching that I hadn't changed it for the `update` case. I've changed that now